### DISCUSSION TITLE
Which branch to use?

### DISCUSSION THREAD

**Discussion by celidos** [open-edge-platform/anomalib#1553] [Q&A]
Hi all!

I want to try my hand at implementing [WinCLIP](https://arxiv.org/abs/2303.14814). With version 1 of the library coming out soon, should I branch off from the `v1` branch?

Btw, is it okay to take the unofficial implementation from **caoyunkang** here https://github.com/caoyunkang/WinClip as a basis (distributed under the MIT license)?

I'm new and haven't contributed to such projects before.

P.S. Thank you very much for your library, it is extremely handy ❤️

**Comment by blaz-r** [comment]
Hello. It's really great to hear you like the library. ❤️ 

I think that you should use the v1 as a base, since it's going to be released very soon.

Regarding the implementation, I think it's fine if you take that one as a basis, but make sure you credit the original author in readme 😄. Also read the [contributing guidelines](https://github.com/openvinotoolkit/anomalib/blob/main/CONTRIBUTING.md).

@samet-akcay could you please confirm all this?

**Reply by samet-akcay** [reply]
Hi @celidos, happy to hear this!

As @blaz-r suggested, you could use v1. However, not sure about WinCLIP implementation. @djdameln is already working on the winclip implementation, which is almost complete. To avoid working on the same thing, I would suggest implementing another model. 

Maybe we could communicate better about upcoming features and models to avoid such duplication

**Reply by blaz-r** [reply]
Oh, I didn't know that. I agree that for model implementations there should be some way of communicating what is in the works. Maybe a draft PR once someone starts implementing.

I think that there are currently no diffusion based models in Anomalib like [DiffusionAD](https://paperswithcode.com/paper/diffusionad-denoising-diffusion-for-anomaly), [DDAD](https://paperswithcode.com/paper/anomaly-detection-with-conditioned-denoising) or [TransFusion](https://paperswithcode.com/paper/transfusion-a-transparency-based-diffusion). They are top 3 in detection AUROC on VisA and very recent, so that would be a very valuable addition.

There was also [M3DM](https://paperswithcode.com/paper/multimodal-industrial-anomaly-detection-via) at CVPR this year for 3D anomaly detection. Even better performing and very recent model for 3D anomaly detection is [3DSR](https://arxiv.org/pdf/2311.01117.pdf), but there is not public code yet.

**Reply by celidos** [reply]
Okay!

I was curious about (maybe list?) of hot features that would be great to add to anomalib, but which no one is working on now (just to avoid duplication)

There is a great awesome-list of recent advances
https://github.com/M-3LAB/awesome-industrial-anomaly-detection

Maybe you have some ideas?

UPD: just noticed your response, thank you! I'll try diffusion models you've mentioned above

**Reply by blaz-r** [reply]
Cool. Addition of diffusion model would be really great, thanks. Both DiffusionAD and DDAD have available code that is MIT licensed. @samet-akcay just to confirm, is any of these models already in development, and do you have any preference which one would be better addition?
